backend/fms_core/services/library.py

```diff
@@ -109,6 +106,13 @@ def convert_library(process, platform, sample_source, container_destination, coo
             derived_samples_destination = []
             volume_ratios = {}
             for derived_sample_source in sample_source.derived_samples.all():
+                library_source_obj = derived_sample_source.library
+                # extra validation
+                if library_source_obj is None:
+                    errors.append(f"Sample {sample_source.name} is not a library or pool of libraries. ")
```

Comment:
This should be reported differently. It means there is a mix of libraries and samples in the pool. I would report it as "Pool {sample_source.name} contains a sample {derived_sample_source.biosample.alias} that is not converted into a library." Since you tested for library already a pool is the only case you'll get here.